### PR TITLE
prov/gni: Fix bug in __gnix_ep_fini_vc.

### LINKED ISSUE #893: rdm_mbw_mr get warnings in __gnix_vec_remove_at
One or more of theses warnings shows up at the end of the test:

```
libfabric:gni:ep_ctrl:__gnix_vec_remove_at():195<warn> No entry exists in __gnix_vec_remove_at
```

### PR DESCRIPTION
**Pull request by e-harvey**
- The vector changes in #887 modified the iterator to move the current index forward after calling _gnix_vec_iter_next causing __gnix_ep_fini_vc to attempt removal of an out-of-bounds entry.
- Updated comments in gnix_vector.h to reflect the gnix_vector api changes.
- Updated __gnix_ep_fini_vc to remove elements at the correct index.

@sungeunchoi @ztiffany 
Fixes #893

### CODE REVIEW

File: prov/gni/src/gnix_ep.c [hunk 1]
```diff
@@ -361,7 +361,7 @@ static inline int __gnix_ep_fini_vc(struct gnix_fid_ep *ep)
 		while ((vc = (struct gnix_vc *)
 				_gnix_vec_iterator_next(&iter))) {
 			_gnix_vec_remove_at(ep->vc_table,
-					    GNIX_VECTOR_ITERATOR_IDX(iter));
+					    GNIX_VECTOR_ITERATOR_IDX(iter) - 1);
```
Comment (ztiffany) [inline prov/gni/src/gnix_ep.c]: If you this route you should do the decrement in this macro instead:

 #define GNIX_VECTOR_ITERATOR_IDX(_iter)    ((_iter).cur_idx - 1)

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by e-harvey** [comment]
The LANL-Cray Jenkins build is failing because the machine cannot fetch from https://github.com/ofi-cray/libfabric-cray/.

**Comment by ztiffany** [comment]
:+1:

**Comment by hppritcha** [comment]
something's going on with NERSC.
bot:retest

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by sungeunchoi** [comment]
👍

**Comment by hppritcha** [comment]
okay looks like SSH key problem
bot:retest

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by hppritcha** [comment]
bot:retest

**Comment by e-harvey** [comment]
Thanks.

**Comment by hppritcha** [comment]
When I try to merge this PR upstream I get segfaults in the criterion test.

**Comment by hppritcha** [comment]
I suspect I'm getting the segfaults because #887 isn't merged upstream yet.  We need to resolve
issue #896 before pushing this PR upstream.

**Comment by e-harvey** [comment]
I will be looking further into #896 this morning.

**Pull request merged**